### PR TITLE
feat(btcpayserver): add support for BTCPayServer

### PR DESCRIPTION
**Pull request by mrfelton**
Allow connecting to an LND instance that is behind BTCPayServer.

### CODE REVIEW

File: app/components/Onboarding/BtcPayServer.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+}) => (
+  <div className={styles.container}>
+    <section className={styles.input}>
+      <label htmlFor="connectionString">Connection String:</label>
+      <textarea
+        type="text"
+        id="connectionString"
+        rows="9"
+        placeholder="BTCPayServer Connection String"
+        className={`${
```
Comment (Empact) [inline app/components/Onboarding/BtcPayServer.js]: Quoting looks unnecessary here

Comment (mrfelton) [reply app/components/Onboarding/BtcPayServer.js]: Could you elaborate please @Empact . I'm not sure I follow

Comment (Empact) [reply app/components/Onboarding/BtcPayServer.js]: You should be able to remove `\`${` bit, and the closing, because then it would resolve down to `undefined` or `styles.error` directly, which don't need quoting afaict.

Comment (mrfelton) [reply app/components/Onboarding/BtcPayServer.js]: Gotcha. You are right - I've altered that. Thanks.

File: app/lnd/lib/lightning.js [hunk 2]
```diff
@@ -16,7 +16,9 @@ process.env.GRPC_SSL_CIPHER_SUITES =
     'ECDHE-ECDSA-AES128-GCM-SHA256',
     'ECDHE-ECDSA-AES256-GCM-SHA384',
     'ECDHE-ECDSA-AES128-CBC-SHA256',
-    'ECDHE-ECDSA-CHACHA20-POLY1305'
+    'ECDHE-ECDSA-CHACHA20-POLY1305',
+    'ECDHE-RSA-AES256-GCM-SHA384',
+    'ECDHE-RSA-AES128-GCM-SHA256'
```
Comment (Empact) [inline app/lnd/lib/lightning.js]: Could you add a comment as to the need for these?

Comment (mrfelton) [reply app/lnd/lib/lightning.js]: I've added a comment to that effect.

File: app/components/Onboarding/BtcPayServer.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            ? styles.error
+            : undefined
+        }
+        ref={input => input}
+        value={connectionString}
+        onChange={event => setConnectionString(event.target.value)}
+      />
+      <p
+        className={`${styles.errorMessage} ${
+          connectionString && !connectionStringIsValid ? styles.visible : undefined
```
Comment (Empact) [inline app/components/Onboarding/BtcPayServer.js]: nit: explicit precedence

File: app/components/Onboarding/BtcPayServer.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        }`}
+      >
+        Invalid connection string.
+      </p>
+      <p
+        className={`${styles.errorMessage} ${
+          connectionString && startLndHostError ? styles.visible : undefined
```
Comment (Empact) [inline app/components/Onboarding/BtcPayServer.js]: nit: explicit precedence

File: app/components/Onboarding/BtcPayServer.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  <div className={styles.container}>
+    <section className={styles.input}>
+      <label htmlFor="connectionString">Connection String:</label>
+      <textarea
+        type="text"
+        id="connectionString"
+        rows="9"
+        placeholder="BTCPayServer Connection String"
+        className={
+          connectionString && (startLndHostError || !connectionStringIsValid)
```
Comment (Empact) [inline app/components/Onboarding/BtcPayServer.js]: nit: explicit precedence

File: app/lnd/config/index.js [hunk 6]
```diff
@@ -58,12 +58,18 @@ if (process.env.NODE_ENV === 'development') {
...
+    const host = store.get('host')
+    const macaroon = store.get('macaroon')
+
+    return {
+      lndPath,
+      configPath: join(appRootPath, 'resources', 'lnd.conf'),
+      rpcProtoPath: join(appRootPath, 'resources', 'rpc.proto'),
+      host: typeof host === 'undefined' ? 'localhost:10009' : host,
+      cert: typeof cert === 'undefined' ? join(userInfo().homedir, loc) : cert,
+      macaroon: typeof macaroon === 'undefined' ? join(userInfo().homedir, macaroonPath) : macaroon
```
Comment (Empact) [inline app/lnd/config/index.js]: Why this change? `false` does not seem valid or meaningful for these.

Comment (mrfelton) [reply app/lnd/config/index.js]: The purpose is that no value is a valid value. For example, if an empty string has been set we should not assume the default values.

File: app/components/Onboarding/BtcPayServer.js (hunk 3 above)
Comment (mrfelton) [reply app/components/Onboarding/BtcPayServer.js]: Can you explain what you mean by this please @Empact ? Thanks.

Comment (Empact) [reply app/components/Onboarding/BtcPayServer.js]: I prefer making precedence explicit anywhere there is potential ambiguity, in this case between `&&` & `?`. e.g.

`(connectionString && !connectionStringIsValid) ? styles.visible : undefined`

**Comment by mrfelton** [comment]
This has been rebased and updated to work with the latest BTCPayServer connection config file format.

File: app/components/Onboarding/BtcPayServer.js (hunk 3 above)
Comment (mrfelton) [reply app/components/Onboarding/BtcPayServer.js]: Our lint rules do not allow for this. Prettier automatically removes the brackets.

**Comment by JimmyMow** [comment]
Tested ACK c247e7a 🎊

**Comment by Empact** [comment]
"BTCPay Server" rather than "BTCPayServer"
https://github.com/btcpayserver/btcpayserver

**Comment by mrfelton** [comment]
I've updated "BTCPayServer" to read "BTCPay Server" everywhere where it is printed in the UI

**Comment by Empact** [comment]
nit: Could reduce the font size on the connection string input - it's significcantly heavier than most other text and limits what can be seen:
screen shot 2018-07-22 at 14 40 52

**Comment by mrfelton** [comment]
Update font style:

**without error:**
screenshot 2018-07-22 20 46 17

**with error:**
screenshot 2018-07-22 20 46 26

**Comment by Empact** [comment]
Tested ACK fd7f447 ⚡️
